pyomo/gdp/plugins/binary_multiplication.py

```diff
@@ -0,0 +1,173 @@
+from .gdp_to_mip_transformation import GDP_to_MIP_Transformation
+from pyomo.common.config import ConfigDict, ConfigValue
+from pyomo.core.base import TransformationFactory
+from pyomo.core.util import target_list
+from pyomo.gdp import Disjunction
+from weakref import ref as weakref_ref
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+@TransformationFactory.register(
+    'gdp.binary_multiplication',
+    doc="Reformulate the GDP as an MINLP by multiplying f(x) <= 0 by y to get f(x) * y <= 0 where y is the binary corresponding to the Boolean indicator var of the Disjunct containing f(x) <= 0.",
```

Comment:
Can we break this into multiple strings to keep within Black's preferred 88-character lines?